python/grass/jupyter/interactivemap.py

```diff
@@ -347,13 +353,161 @@ def add_layer_control(self, **kwargs):
         else:
             self.layer_control_object = self._ipyleaflet.LayersControl(**kwargs)
 
+    def draw_computational_region(self):
+        """
+        Allow Users to draw the computational region and modify it.
+        """
+        import ipywidgets as widgets
+        from ipyleaflet import Rectangle, WidgetControl
+
+        region_mode_button = widgets.ToggleButton(
+            icon="square",
+            value=False,
+            tooltip="Click to add computational region",
+            button_style="info",
+            layout=widgets.Layout(width="33px", margin="0px 0px 0px 0px"),
+        )
+
+        save_button = widgets.Button(
+            icon="save",
+            button_style="success",
+            tooltip="Click to save region",
+            layout=widgets.Layout(width="33px", margin="0px 0px 0px 0px"),
+            disabled=True,
+        )
+
+        bottom_output_widget = widgets.Output(
+            layout={
+                "width": "100%",
+                "max_height": "300px",
+                "max_width": "300px",
+                "overflow": "auto",
+                "display": "none",
+            }
+        )
+
+        region_coordinates = {}
+        reprojected_region = {
+            "north": None,
+            "south": None,
+            "east": None,
+            "west": None,
+        }
+        rectangle = None
+        save_button_control = None
+
+        def update_output():
+            with bottom_output_widget:
+                bottom_output_widget.clear_output()
+                print("Current Bounds:")
+                print(f"North: {reprojected_region['north']}")
+                print(f"South: {reprojected_region['south']}")
+                print(f"East: {reprojected_region['east']}")
+                print(f"West: {reprojected_region['west']}")
+
+        def on_rectangle_change(value):
+            nonlocal region_coordinates, reprojected_region
+            latlon_bounds = value["new"][0]
+            region_coordinates = {
+                "north": latlon_bounds[2]["lat"],
+                "south": latlon_bounds[0]["lat"],
+                "east": latlon_bounds[2]["lng"],
+                "west": latlon_bounds[0]["lng"],
+            }
+            from_proj = "+proj=longlat +datum=WGS84 +no_defs"
+            to_proj = get_location_proj_string()
+            reprojected_region = reproject_region(
+                region_coordinates, from_proj, to_proj
+            )
+            update_region(
```

Comment:
`update_region` should probably take as input a dictionary (similarly to `reproject_region`)